src/motor_control/src/motor_control_node.cpp

```diff
@@ -58,119 +55,141 @@ class MotorControlNode : public rclcpp::Node {
     percentPower = std::clamp(percentPower, (float)(-1), (float)(1));
-
     int32_t data = percentPower * 100000; // Convert from percent power to a signed 32-bit integer
 
     send_can(id + 0x00000000, data); // ID does NOT need to be modified to signify this is a duty cycle command
     this->current_msg[id] = std::make_tuple(id + 0x00000000, data); // update the hashmap
-    // RCLCPP_INFO(this->get_logger(), "Setting the duty cycle of CAN ID: %u to %f", id, percentPower);
+    // RCLCPP_INFO(this->get_logger(), "Setting the duty cycle of CAN ID: %u to %f", id, percentPower); // Print Statement
   }
 
   // Set the velocity of the motor in RPM (Rotations Per Minute)
   void vesc_set_velocity(uint32_t id, int rpm) {
     int32_t data = rpm;
 
-    send_can(id + 0x00000300, data); // ID must be modified to signify this is an RPM command
-    this->current_msg[id] = std::make_tuple(id + 0x00000300, data);                  // update the hashmap
-    // RCLCPP_INFO(this->get_logger(), "Setting the RPM of CAN ID: %u to %d", id, rpm); // Print to the terminal
+    send_can(id + 0x00000300, data); // ID must be modified to signify this is a RPM command
+    this->current_msg[id] = std::make_tuple(id + 0x00000300, data); // update the hashmap
+    // RCLCPP_INFO(this->get_logger(), "Setting the RPM of CAN ID: %u to %d", id, rpm); // Print Statement
   }
 
-  // Set the position of the motor in _____ (degrees? encoder counts?)
+  // Set the position of the motor in degrees // TODO: Position control has not been tested yet!
   void vesc_set_position(uint32_t id, int position) {
-    // TODO: Implement this method!
+    int32_t data = position * 1000000;
+
+    send_can(id + 0x00000400, data); // ID must be modified to signify this is a position command
+    this->current_msg[id] = std::make_tuple(id + 0x00000400, data); // update the hashmap
+    // RCLCPP_INFO(this->get_logger(), "Setting the position of CAN ID: %u to %d", id, position); // Print Statement
   }
 
-  // Set the current draw of the motor in amps
+  // Set the current draw of the motor in amps // TODO: Current control has not been fully tested yet!
   void vesc_set_current(uint32_t id, float current) {
-    // TODO: Implement this method!
+    int32_t data = current * 1000; // Current is measured in amperage
+
+    send_can(id + 0x00000100, data); // ID must be modified to signify this is a current command
+    this->current_msg[id] = std::make_tuple(id + 0x00000100, data); // update the hashmap
+    // RCLCPP_INFO(this->get_logger(), "Setting the current of CAN ID: %u to %f amps", id, current); // Print Statement
   }
 
   // Get the current duty cycle of the motor
   float vesc_get_duty_cycle(uint32_t id) {
     if (std::chrono::steady_clock::now() - this->can_data[id].timestamp < this->threshold) {
       return this->can_data[id].dutyCycle;
     } else {
-      return -1; // Return -1 if the data is stale
+      return -999; // Return -999 if the data is stale
```

Comment:
Is using a sentinel value the best solution here?

Reply:
fixed using std::optional